curriculum/challenges/english/25-front-end-development/lab-sentence-maker/66c057041df6394ca796bf33.md

```diff
@@ -124,7 +124,7 @@ You should use the correct story format for the first story: `"Once upon a time,
 ```js
 const _initialValues = {}
 for (const name of ['adjective', 'noun', 'noun2', 'place', 'verb', 'adjective2']) {
-    const match = code.match(new RegExp(String.raw`${name}\s*=\s*('|"|${'`'})(?<${name}>.*)\1;?\s*`));
+    const match = code.match(new RegExp(String.raw`${name}\s*=\s*('|"|${'`'})(?<${name}>.*?)\1(?:,|;|\s|$)`));
```

Comment:
```suggestion
    const match = code.match(new RegExp(String.raw`${name}\s*=\s*('|"|${'`'})(?<${name}>.*?)\1(?:,|;|\s|$)`, 'm'));
```

maybe?
if you use `$` without `m` it matches only end of string, if you use it with `m` flag it matches end of line

Reply:
Does the current regex match incorrectly, either matching something it shouldn't, or not matching something it should?

I'm not sure what code to test it with to see the difference. Can you give an example?

But I'm fine with making the change if needed. Just asking for clarification.

Reply:
not really, the `\s` matches a `\n` too, so it doesn't matter